dynawo/sources/Models/CPP/ModelNetwork/DYNModelShuntCompensator.cpp

```diff
@@ -211,10 +211,10 @@ ModelShuntCompensator::evalZ(const double& t) {
   z_[isAvailableNum_] = isAvailable() ? 1. : 0.;
   z_[currentSectionNum_] = getCurrentSection();
 
-  if (modelBus_->getConnectionState() == OPEN)
+  State currState = static_cast<State>(static_cast<int>(z_[connectionStateNum_]));
+  if (modelBus_->getSwitchOff() && currState == CLOSED && getConnected() != OPEN)
```

Comment:
Why is this test more complex than the others?

Reply:
Because of the automatons that could connect to the state

Reply:
What would be a scenario in which a shunt connected to a switchedOff bus should stay connected, even if closed by an automaton? Maybe we should play on the "Available" signal here

Reply:
That's a good question and I don't remember the scenario at play here. I would have to go back in debug to understand again why all those conditions were needed. For the available maybe but I never looked at it

Reply:
If I remember well, when the "Available" signal is off, the automaton does not consider the shunt for connection/disconnection. If this is correct, the shunt should not be available anymore when the bus is switchedOff